tests/lib/util/config-comment-parser.js

```diff
@@ -0,0 +1,1908 @@
+                    value: false,
+                    comment
+                }
+            });
+        });
+
+        it("should parse Boolean config with two items", () => {
+            const code = "a: true b:false";
+            const result = commentParser.parseBooleanConfig(code, comment);
+
+            assert.deepStrictEqual(result, {
+                a: {
+                    value: true,
+                    comment
+                },
+                b: {
+                    value: false,
+                    comment
+                }
+            });
+        });
+
+        it("should parse Boolean config with two comma-separated items", () => {
+            const code = "a: true, b:false";
+            const result = commentParser.parseBooleanConfig(code, comment);
+
+            assert.deepStrictEqual(result, {
+                a: {
+                    value: true,
+                    comment
+                },
+                b: {
+                    value: false,
+                    comment
+                }
+            });
+        });
+
+        it("should parse Boolean config with two comma-separated items and no values", () => {
+            const code = "a , b";
+            const result = commentParser.parseBooleanConfig(code, comment);
+
+            assert.deepStrictEqual(result, {
+                a: {
+                    value: false,
+                    comment
+                },
+                b: {
+                    value: false,
+                    comment
+                }
+            });
+        });
+    });
+
+    describe("parseListConfig", () => {
+
+        it("should parse list config with one item", () => {
+            const code = "a";
+            const result = commentParser.parseListConfig(code);
+
+            assert.deepStrictEqual(result, {
+                a: true
+            });
+        });
+
+        it("should parse list config with two items", () => {
+            const code = "a, b";
+            const result = commentParser.parseListConfig(code);
+
+            assert.deepStrictEqual(result, {
+                a: true,
+                b: true
+            });
+        });
+
+        it("should parse list config with two items and exta whitespace", () => {
+            const code = "  a , b  ";
+            const result = commentParser.parseListConfig(code);
+
+            assert.deepStrictEqual(result, {
+                a: true,
+                b: true
+            });
+        });
+    });
+
+    //     it("should report a violation when the report is right before the comment", () => {
+    //         const code = " /* eslint-disable */ ";
+
+    /*
+     *         linter.defineRule("checker", context => ({
+     *             Program() {
+     *                 context.report({ loc: { line: 1, column: 0 }, message: "foo" });
+     *             }
+     *         }));
+     *         const problems = linter.verify(code, { rules: { checker: "error" } });
```

Comment:
The refactoring LGTM, but I think we should remove these commented tests (or uncomment them if applicable).

Reply:
Also, I'm guessing all of those tests were copied from linter.js test file, but I'm not seeing any removals in that file.

Reply:
Right - do we want to duplicate tests? I wonder if it would be clearer for the future to leave the tests in linter and just add new unit tests for config-comment-parser?

Reply:
I just copied those there temporarily to help me figure out the best tests to write for this module. The tests needs to stay in `linter.js` because they test the interpretation of the comments along with the parsing. I'll just remove these.